exonum/src/blockchain/mod.rs

```diff
@@ -52,6 +53,92 @@ mod schema;
 #[cfg(test)]
 pub mod tests;
 
+/// Container for transactions allowing to look them up by hash digest.
+///
+/// By default, transaction caches are *ephemeral*; they are not saved on node restart.
+/// However, Exonum nodes do have an ability to save uncommitted transactions to the persistent
+/// cache (although this API is not stable). Reading transactions from such a cache is possible
+/// via [`PersistentCache`].
+///
+/// [`PersistentCache`]: struct.PersistentCache.html
+pub trait TransactionCache {
+    /// Gets a transaction from this cache. `None` is returned if the transaction is not
+    /// in the cache.
+    fn get_transaction(&self, hash: Hash) -> Option<Verified<AnyTx>>;
+
+    /// Checks if the cache contains a transaction with the specified hash.
+    ///
+    /// The default implementation calls `get_transaction()` and checks that the returned
+    /// value is `Some(_)`.
+    fn contains_transaction(&self, hash: Hash) -> bool {
+        self.get_transaction(hash).is_some()
+    }
+}
+
+/// Cache that does not contain any transactions.
+impl TransactionCache for () {
+    fn get_transaction(&self, _hash: Hash) -> Option<Verified<AnyTx>> {
+        None
+    }
+
+    fn contains_transaction(&self, _hash: Hash) -> bool {
```

Comment:
This implementation seems redundant ? And if you'd add a default implementation for the `get_transaction` method like:
```rust
fn get_transaction(&self, hash: Hash) -> Option<Verified<AnyTx>> {
    None
}
```
You could write just
```rust
impl TransactionCache for () { }
```